tests/frameworks/include/TestFramework.h

```diff
@@ -134,27 +206,45 @@
-                                }                                                                                   \
-                                                                                                                    \
-                                for (const auto& item : values2) {                                                  \
-                                    combined.push_back(item);                                                       \
-                                }                                                                                   \
-                                                                                                                    \
-                                return ::testing::ValuesIn(combined);                                               \
-                                                                                                                    \
-                            }())
+#define WIN32_DISABLED_INSTANTIATE_TEST_CASE_P(instantiationName, testClass, values, disabledValues) \
+    INSTANTIATE_TEST_CASE_P(instantiationName, testClass, []() {                                     \
+        ::testing::internal::ParamGenerator<testClass::ParamType> values1 = values;                  \
+        ::testing::internal::ParamGenerator<testClass::ParamType> values2 = disabledValues;          \
+        std::vector<testClass::ParamType> combined;                                                  \
+        for (const auto& item : values1) {                                                           \
+            combined.push_back(item);                                                                \
+        }                                                                                            \
+                                                                                                     \
+        for (const auto& item : values2) {                                                           \
+            combined.push_back(item);                                                                \
+        }                                                                                            \
+                                                                                                     \
+        return ::testing::ValuesIn(combined);                                                        \
+                                                                                                     \
+    }())
 #endif
 
 // OSX defines for compatibility
 #ifndef boolean
 #define boolean Boolean
-#endif
\ No newline at end of file
+#endif
+
+#ifndef GET_CURRENT_TEST_DIRECTORY
+#define GET_CURRENT_TEST_DIRECTORY                                         \
+    []() {                                                                 \
+        std::string tempBuffer;                                            \
```

Comment:
out of convenience / laziness mostly. Since lambdas are anonymous, I don't need to worry about ODR violations or weird storage concerns as much. I believe I could define a static function under the ifndef and be fine though if you think that is more clear?